multiqc/modules/odgi/odgi.py

```diff
@@ -0,0 +1,202 @@
+            sum_of_path_nodes_distances_nucleotide_space.update(
+                {sample_name: float(file_stats['Sum_of_path_nodes_distances {}'.format(sample_name)]['In_nucleotide_space'])})
+
+        metrics_lineplot_config = {
+            'id': 'odgi_metrics_lineplot',
+            'ylab': 'Count',
+            'categories': True,
+            'yDecimals': False,
+            'logswitch': True,
+            'title': 'Odgi: Odgi metrics'
+        }
+
+        self.add_section(
+            name='Odgi metrics',
+            anchor='odgi-stats',
+            description='The odgi metrics section',
+            helptext='<b>sum-of-path-node-distances</b>:<br>For each path we iterate from node to node and count the node distance of nodes on the pangenome '
+                     'level normalized by the path length. If a node is reversed, we count the node distance twice.<br><br>'
+                     '<b>sum-of-path-nucleotide-distances</b>:<br>'
+                     'For each path we iterate from node to node and count the nucleotide distance of nodes on the pangenome level normalized by the path '
+                     'length. If the node is reversed, we count the nucleotide distance twice.<br><br>'
+                     '<b>mean-links-length</b>:<br>For each path we iterate from node to node and count the node distance of nodes within the same path only! '
+                     'We then normalized by the path length.',
+            plot=linegraph.plot({'in_node_space_mean': mean_links_length_nodes_space,
+                                 'in_nucleotide_space_mean': mean_links_length_nucleotide_space,
+                                 'in_node_space_sum': sum_of_path_nodes_distances_nodes_space,
+                                 'in_nucleotide_space_sum': sum_of_path_nodes_distances_nucleotide_space
+                                 }, pconfig=metrics_lineplot_config)
+        )
+
+    def extract_sample_name(self, file_name):
+        """
+        Extracts the sample name from a given file name.
+        Expects and returns one of seqwish, smooth or cons@*
+        """
+        if 'cons@' in file_name:
+            file_name = file_name.split('.')
+            consensus_identifier = list((e for e in file_name if 'cons@' in e))
+            try:
+                return consensus_identifier[0]
+            except IndexError:
+                log.error('Unknown file name {}: File name must either contain seqwish, smooth or cons@!'.format(file_name))
+        elif 'smooth' in file_name:
+            return 'smooth'
+        elif 'seqwish' in file_name:
+            return 'seqwish'
+
+    def compress_stats_data(self, sample_name, stats, mean_links_length, sum_of_path_nodes_distances) -> dict:
+        """
+        Compress odgi stats into a single dictionary to visualize.
+        """
+        return {
+            sample_name: {
+                'General stats {}'.format(sample_name): {
+                    'Length': float(stats[0]),
```

Comment:
Again this is very brittle. Can we not use the column headers in the file instead?